### DISCUSSION TITLE
How to read a base64 image string into Magick::Image Object?

### DISCUSSION THREAD

**Discussion by Haidar0096** [ImageMagick/ImageMagick#5615] [General] [answered]
I tried
```c++
Blob blob;
        blob.base64(sourceImage); // sourceImage is a base64 string representation of my image
        Image image;
        image.size("100x100");
        image.read(blob);
```

But this fails with exception:
```
Caught exception: demo: zero-length blob not permitted `' @ error/blob.c/BlobToImage/448
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
```

How can I read a base64 image properly?

Note: The reason I want to do this is that I want to call image magick library c++ functions from Flutter code, so I need a way to pass data between dart and c++. If there is better way of doing this, please tell me as well.

**Comment by dlemstra** [comment, marked as answer]
Can you share your `base64` string? It looks like this is not a valid string.

**Reply by Haidar0096** [reply]
Thanks for your reply, I generated it from an image using [this website](https://www.base64-image.de/), here it is:

https://justpaste.it/94y0d  (github won't let me paste it here because it is very long)

**Reply by dlemstra** [reply]
That is not a valid base64 string. It starts with `data:image/png;base64,`

**Reply by Haidar0096** [reply]
You are right, I thought the magick decoder can decode it as it is.

I removed the prefixes and it worked, thank you :-)
